src/Evolution/Systems/Burgers/Observe.hpp

```diff
@@ -4,141 +4,176 @@
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
-namespace Burgers {
-namespace Actions {
+/// \cond
+// IWYU pragma: no_forward_declare Tensor
+namespace Tags {
+struct Time;
+}  // namespace Tags
+/// \endcond
 
-namespace observe_detail {
-using reduction_datum = Parallel::ReductionDatum<double, funcl::Plus<>,
-                                                 funcl::Sqrt<funcl::Divides<>>,
-                                                 std::index_sequence<1>>;
-using reduction_datums =
-    tmpl::list<Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
-               Parallel::ReductionDatum<size_t, funcl::Plus<>>,
-               reduction_datum>;
+namespace Burgers {
+namespace Events {
+template <typename EventRegistrars>
+class Observe;
 
-}  // namespace observe_detail
+namespace Registrars {
+using Observe = Registration::Registrar<Events::Observe>;
+}  // namespace Registrars
 
 /*!
- * \brief Temporary action for observing volume and reduction data
+ * \brief Observe the Burgers system
  *
- * A few notes:
- * - Writes the solution and error in \f$U\f$ to disk.
- * - The RMS error of \f$U\f$ is written to disk.
+ * Writes volume quantities:
+ * - `InertialCoordinates`
+ * - `U`
+ * - `ErrorU` = \f$U - \text{analytic solution}\f$
+ *
+ * Writes reduction quantities:
+ * - `Time`
+ * - `NumberOfPoints` = total number of points in the domain
+ * - `ErrorU` = \f$\operatorname{RMS}(U - \text{analytic solution})\f$
+ *   over all points
+ *
+ * \warning Currently, only one observation event can be triggered at
+ * a given time.  Causing multiple events to run at once will produce
+ * unpredictable results.
  */
-struct Observe {
-  struct ObserveNSlabs {
-    using type = size_t;
-    static constexpr OptionString help = {"Observe every Nth slab"};
-  };
-  struct ObserveAtT0 {
-    using type = bool;
-    static constexpr OptionString help = {"If true observe at t=0"};
-  };
-
-  using const_global_cache_tags = tmpl::list<ObserveNSlabs, ObserveAtT0>;
+template <typename EventRegistrars = tmpl::list<Registrars::Observe>>
+class Observe : public Event<EventRegistrars> {
+ private:
+  using reduction_datum =
+      Parallel::ReductionDatum<double, funcl::Plus<>,
+                               funcl::Sqrt<funcl::Divides<>>,
+                               std::index_sequence<1>>;
+  using reduction_datums =
+      tmpl::list<Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+                 Parallel::ReductionDatum<size_t, funcl::Plus<>>,
+                 reduction_datum>;
+
+ public:
+  /// \cond
+  explicit Observe(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(Observe);  // NOLINT
+  /// \endcond
+
+  using options = tmpl::list<>;
+  static constexpr OptionString help =
+      "Observes the Burgers field U and its error.\n"
+      "\n"
+      "Writes volume quantities:\n"
+      " * InertialCoordinates\n"
```

Comment:
Can you add an brief description after a `:`what „Inertial“ means here? Perhaps „The coordinates in which the Burgers equation is formulated“ (if that defines it well enough). Reason is that other systems will use multiple coordinate systems and when we output them we should make clear what they are. And „Inertial“ is not self-explanatory.

Reply:
Explaining dual frames is not the responsibility of this class.  I would offer to link to the documentation on the subject, but there doesn't appear to be any.

Reply:
I agree with @wthrowe. It's likely that some of these things will be updated as dual frames are implemented. This is certainly not the place to explain dual frames/ALE/moving mesh.

Reply:
I agree InertialCoordinates should be documented, but not here.  I see @wthrowe opened #1409 to address this, so let's move one